rule/expr.go

```diff
@@ -147,7 +148,7 @@ func FlattenExpr(e bzl.Expr) bzl.Expr {
 		return true
 	}
 
-	if ps.generic != nil {
+	if ps.generic != nil && !reflect.ValueOf(ps.generic).IsNil() {
```

Comment:
Interface values should almost always be non-nil or untyped nil. This check implies a typed nil is a valid value, and I'm worried that's very error-prone.